icon-project/governance
# IS-838: Migrate system configuration states from governance score to iconservice

### Pre-Definition
- INV: Icon Network Value. 
  - service_config
  - step_price
  - step_costs
  - max_step_limits
  - revision_code
  - revision_name
  - score_black_list
  - import_white_list
- GS: Governance Score
- IS: Icon Service

### Major Changes
- Migrate INV to icon-service using migration API and remove all INVs which have been recorded on GS.
- Read and Update INV using 'get_icon_network_value' and 'set_icon_network_value' API.
- Remove 'on_install' logic: This method won't be called by the icon-service. Therefore, remove this method to reduce confusion.
- Remove '_remove_import_white_list' method: This method is not called by any methods. (Already has been deprecated before)
- Remove 'StepCosts' class
- Remove deployer related variable and method: This value is not used anymore neither IS nor GS. So even do not migrate.
- Remove properties: properties were called by icon-service before, but after migration, do not call these properties.

### How To Tests GS (governance/tests/)
To test this feature (in governance/tests), you must install below packages with a specific branch (i.e. pip install -e) 
- t-bears:  IS-838-temp-branch-for-testing-governance-migration 
- icon-service: IS-1013-alter-tests-from-unittest-to-pytest **(Caution!! Not IS-838 branch)**
- icon-sdk: IS-720-support-block-v0.3 branch and change teh version to 1.3.0

**note:** You can check the each INV data's format in confluence 'Migrate system configuration states from GS to IS'
